api/src/models/JobMetadata.ts

```diff
@@ -25,6 +29,22 @@ class JobMetadata {
         }
     }
 
+    ingestLockfile(job) {
```

Comment:
This method shouldn't need to take a job parameter; I believe it can use this.job instead, since the job was passed to the constructor and saved internally.